### PR TITLE
Update documentation and expected module usage behaviour

### PR DESCRIPTION
**Pull request by bkuebler**
#### Pull Request (PR) description
The current module documentation in code has an old style and a REFERENCE.md is missing.
This PR updates all documentation in this module to the new style and provides the REFERENCE.md file.
Also the assumption of module usage was corrected with add `assert_private()` to several classes. At the end
i also added more CI tests, that the test coverage is now increased to 98.10%

#### This Pull Request (PR) fixes the following issues
none

### CODE REVIEW

File: manifests/transport.pp [hunk 1]
```diff
@@ -1,51 +1,48 @@
...
-#     
-#   }
+# @see https://www.postfix.org/transport.5.html
 #
 define postfix::transport (
-  Optional[String]          $destination = undef,
-  Optional[String]          $nexthop=undef,
-  Optional[Stdlib::Absolutepath] $file=undef,
-  Enum['present', 'absent'] $ensure='present'
+  Enum['present', 'absent']      $ensure      = 'present'
```
Comment (smortex) [inline manifests/transport.pp:42]:
```suggestion
  Enum['present', 'absent']      $ensure      = 'present',
```

**Review by smortex — CHANGES_REQUESTED** [review]
That looks awesome <3

File: manifests/params.pp [hunk 2]
```diff
@@ -22,4 +26,5 @@
   Optional[String] $aliasesseltype,
   Optional[String] $seltype,
 ) {
+  assert_private()
```
Comment (smortex) [inline manifests/params.pp:29]: This also break compatibility.  However this class is empty so maybe we can delete it completely?

The `foo::param` class predates Hiera and was used to setup os-specific variables. Then the `foo` class could inherit from it to get that default value and allow the user to override it.  This `params` class with parameters is unusual and probably something that even predates that, and we can see direct use of `postfix::params::foo` here and there in the code.

Maybe for now we can ignore this smell and just remove the assert_private to fix CI.

File: manifests/satellite.pp [hunk 3]
```diff
@@ -33,6 +10,7 @@
   $masquerade_domains    = undef,
   $masquerade_exceptions = undef,
 ) {
+  assert_private()
```
Comment (smortex) [inline manifests/satellite.pp:13]: This breaks CI.  A privte class with parameters is weird… probably not that private, or maybe a design flaw :smile:

File: manifests/mta.pp [hunk 4]
```diff
@@ -36,6 +10,7 @@
   Optional[Array[String[1]]]                            $masquerade_domains  = undef,
   Optional[Array[String[1]]]                            $masquerade_exceptions = undef,
 ) {
+  assert_private()
```
Comment (smortex) [inline manifests/mta.pp:13]: Same here as `postfix::satellite`

File: manifests/mailman.pp [hunk 5]
```diff
@@ -1,17 +1,13 @@
...
+# @api private
 #
-# /!\ Do not include this class directly,
-# use mailman => true in the postfix top class!
-#
-#   class { 'postfix':
-#     mailman => true,
-#   }
 class postfix::mailman {
+  assert_private()
```
Comment (smortex) [inline manifests/mailman.pp:9]: Same here, this break CI

**Review by smortex — COMMENTED** [review]
Switching some classes to private breaks CI.  A quickfix is to not make the classes private, but if you feel like tweaking the tests, this idea would be to remove the tests that operate directly on the private classes and add them to the tests that manipulate the class that manipulate these private classes.

File: manifests/augeas.pp [hunk 6]
```diff
@@ -1,7 +1,12 @@
-#class postfix::augeas
-#  This class provides the augeas lenses used by the postfix class
+# @summary Provides augeas lenses for postfix files
+#
+# This class provides the augeas lenses used by the postfix class
+#
+# @api private
 #
 class postfix::augeas {
+  assert_private()
```
[the remark this answers is not in the document]
Comment (bkuebler) [reply manifests/augeas.pp:8]: CI Tests reworked

File: manifests/params.pp (hunk 2 above)
Comment (bkuebler) [reply manifests/params.pp]: CI Tests reworked

File: manifests/satellite.pp (hunk 3 above)
Comment (bkuebler) [reply manifests/satellite.pp:13]: CI Tests reworked

File: manifests/mailman.pp (hunk 5 above)
Comment (bkuebler) [reply manifests/mailman.pp:9]: CI Tests reworked

File: manifests/mta.pp (hunk 4 above)
Comment (bkuebler) [reply manifests/mta.pp:13]: CI Tests reworked

**Review by smortex — APPROVED** [review]
Awesome! I added a commit to fix a few really minor style inconsistencies.  Feel free to adjust this if need be.  As far as I am concerned this in :+1: 

Thanks!

**Comment by smortex** [comment]
Nobody complained, so let's merge this.  Thanks

**Pull request merged**